prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -57,7 +60,13 @@ def __call__(self, a: float, b: float) -> float:
         return a * b
 
 
-class GetMarkets(Function):
+class MarketFunction(Function):
+    def __init__(self, market_type: MarketType) -> None:
+        self.market_type = market_type
+        super().__init__()
```

Comment:
@kongzii Do you have any thoughts on whether `self.market_type` should be set based on env var vs programatically, as I have done in this PR?

@gabrielfior and I are split on it.

From what I see pros of having it being programatic:
- consistent with how we deploy/benchmark/monitor agents in PMAT
- easy to parametrize on it in pytest

Pros of env var:
- consistent with other config vars like wallet address/pkey

Reply:
Some additional thoughts:
- If we do have an env like `MARKET_TYPE=OMEN`, we could restrict the tools the agent has access to by having some kind of mapping (market_type -> list[Function]), or even fancier, we could decorate the functions that belong to Omen as @OmenFunction, so that we can fetch all functions decorated like this programatically and pass them to the agent, depending on the env variable.
- Happy with both approaches here.

Reply:
> If we do have an env like MARKET_TYPE=OMEN, we could restrict the tools the agent has access to by having some kind of mapping (market_type -> list[Function])

Why that wouldn't be possible with a programmatic approach?

>  Do you have any thoughts on whether self.market_type should be set based on env var vs programatically, as I have done in this PR?

I'm not sure you guys will like my point of view, as it brings the most work, but here it goes! 😄 

Personally, I would do kind of both, in a way as in the PMAT repository:

- Remove all the `load_dotenv` callings, because modifying state just by importing files isn't nice
- Create single config class as in PMAT [here](https://github.com/gnosis/prediction-market-agent-tooling/blob/513f85cc55da6347758be822c1e16f1c5fd6b5fe/prediction_market_agent_tooling/config.py#L18), so we have all the configuration at one place, because if `os.environ.get` is used too much across the code base, then it's hard to keep track of all the configs available
- Initialise and use that class directly where it make sense or somewhere beforehand and just pass the concrete input arguments

In this case, it makes sense to me that `MarketFunction` accepts  `market_type: MarketType` as the init argument instead of deciding about it quietly in the background (another benefit I can see is that in the future, there could be multiple markets available to agent and that wouldn't be possible if `MarketFunction` behaviour is baked into environment variables)

Reply:
> Create single config class as in PMAT [here](https://github.com/gnosis/prediction-market-agent-tooling/blob/513f85cc55da6347758be822c1e16f1c5fd6b5fe/prediction_market_agent_tooling/config.py#L18)

Nooo, I want to remove that! (https://github.com/gnosis/prediction-market-agent-tooling/issues/71) 😄. But actually for a particular agent I think it makes much more sense than in the PMAT repo, as the keys required are fixed and known

Reply:
> But actually for a particular agent I think it makes much more sense than in the PMAT repo, as the keys required are fixed and known

Yep, not a great choice for tooling repo, but for the agent it sounds good to me

Reply:
Anyway, no need to do that in this PR. TLDR is that I like programic `__init__(self, market_type: MarketType)` more here :-D Doing it inside of __init__ from env vars will be error prone.

Reply:
It's a good point though - I've imported APIKeys here to replace our manual .env loading. So if it is removed from PMAT it can live on in this repo 😄

Reply:
I just remembered that we already have something in PMA repo too: https://github.com/gnosis/prediction-market-agent/blob/46d0ddae565909d8dd617e6448cbf40a4dc8c4d9/prediction_market_agent/utils.py#L11

It subclasses PMAT's API keys, so the same stuff is available in PMA's APIKeys as well, but once they are removed in PMAT, should be easier to just add required stuff here.

Reply:
Ah nice, I've switched to that here https://github.com/gnosis/prediction-market-agent/pull/63